File: Section Reports/Layout/SubReport/VB.NET/SubReport/rptDSRelationChildMain.vb
Public Class rptDSRelationChildMain
	Inherits GrapeCity.ActiveReports.SectionReport
	Dim _drc() As DataRow
	Dim _cnt As Integer
	Dim _rpt As rptDSRelationChildSub
	Public Sub New()
		MyBase.New()
		InitializeComponent()
	End Sub
	Private Sub Detail_Format(ByVal sender As Object, ByVal e As EventArgs) Handles Detail.Format
		_drc = CType(DataSource, DataRow())
		''Gets an array of child DataRow objects that are associated with the name of the relationship "ProductsOrderDetails" and set the DataSource of the sub-report.
		''
		If (_cnt < _drc.Length) Then
			_rpt.DataSource = _drc(_cnt).GetChildRows("ProductsOrderDetails")
		End If
		_cnt = _cnt + 1
	End Sub
	Private Sub rptDSRelationChildMain_ReportStart(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.ReportStart
		_rpt = New rptDSRelationChildSub
		SubReport1.Report = _rpt
	End Sub
End Class

File: Section Reports/Layout/SubReport/C#/SubReport/rptDSRelationChildMain.cs
using System;
using System.Data;
namespace GrapeCity.ActiveReports.Samples.SubReport
{
	/// <summary>
	/// Description of the outline of the rptDSRelationChildMain.
	/// 
	/// </summary>
	public partial class rptDSRelationChildMain : GrapeCity.ActiveReports.SectionReport
	{
		rptDSRelationChildSub _rpt;
		int _cnt;
		public rptDSRelationChildMain()
		{
			//
			// ActiveReport Designer support is required.
			// 
			//
			InitializeComponent();
		}
		private void Detail_Format(object sender, EventArgs e)
		{
			DataRow[] drc = ((DataRow[])DataSource);
			//Gets an array of child DataRow objects that are associated with the name of the relationship "ProductsOrderDetails" and set the DataSource of the sub-report.
			//
			_rpt.DataSource = drc[_cnt].GetChildRows("ProductsOrderDetails");
			SubReport1.Report = _rpt;
			_cnt++;
		}
		private void rptDSRelationChildMain_ReportStart(object sender, EventArgs e)
		{
			_cnt = 0;
			_rpt = new rptDSRelationChildSub();
		   
		}
	}
}
